pkg/profile/artifact.go

```diff
@@ -7,54 +7,181 @@ import (
+			notReadyConditions = append(notReadyConditions, condition)
+		}
+	}
+
+	if kustomizeRes != nil {
+		if condition := getNotReadyCondition(kustomizeRes.Status.Conditions); condition != (metav1.Condition{}) {
+			notReadyConditions = append(notReadyConditions, condition)
+		}
+	}
+	return notReadyConditions
+}
+
+func (p *Profile) getResources(ctx context.Context) (bool, *sourcev1.GitRepository, *helmv2.HelmRelease, *kustomizev1.Kustomization, error) {
+	gitRes, helmRes, kustomizeRes, err := p.makeArtifacts()
+	if err != nil {
+		return false, nil, nil, nil, err
+	}
+	gitResExists := true
+	helmResExists := true
+	kustomizeResExists := true
+	if gitRes != nil {
+		gitResExists, err = p.getResourceIfExists(ctx, gitRes)
+		if err != nil {
+			return false, nil, nil, nil, err
+		}
+	}
+
+	if helmRes != nil {
+		helmResExists, err = p.getResourceIfExists(ctx, helmRes)
+		if err != nil {
+			return false, nil, nil, nil, err
+		}
+	}
+
+	if kustomizeRes != nil {
+		kustomizeResExists, err = p.getResourceIfExists(ctx, kustomizeRes)
+		if err != nil {
+			return false, nil, nil, nil, err
+		}
+	}
+	return gitResExists && helmResExists && kustomizeResExists, gitRes, helmRes, kustomizeRes, nil
+}
+
+func getNotReadyCondition(conditions []metav1.Condition) metav1.Condition {
+	for _, condition := range conditions {
+		if condition.Type == "Ready" && string(condition.Status) != "True" {
+			return condition
+		}
+	}
+	return metav1.Condition{}
+}
+
+func (p *Profile) getResourceIfExists(ctx context.Context, res client.Object) (bool, error) {
+	if err := p.client.Get(ctx, client.ObjectKey{Name: res.GetName(), Namespace: res.GetNamespace()}, res); err != nil {
+		if apierrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("failed to get resource: %w", err)
+	}
+
+	return true, nil
+}
+
 // MakeArtifacts creates and returns a slice of runtime.Object values, which if
 // applied to a cluster would deploy the profile as a HelmRelease.
 func (p *Profile) MakeArtifacts() ([]runtime.Object, error) {
 	objs := []runtime.Object{}
-	gr, err := p.makeGitRepository()
+	gitRes, helmRes, kustomizeRes, err := p.makeArtifacts()
 	if err != nil {
 		return nil, err
 	}
 
-	hr, err := p.makeHelmRelease()
+	objs = append(objs, gitRes)
+
+	if helmRes != nil {
+		objs = append(objs, helmRes)
+	} else {
+		objs = append(objs, kustomizeRes)
+	}
+	return objs, nil
+}
+
+func (p *Profile) makeArtifacts() (*sourcev1.GitRepository, *helmv2.HelmRelease, *kustomizev1.Kustomization, error) {
+	gitRes, err := p.makeGitRepository()
 	if err != nil {
-		return nil, err
+		return nil, nil, nil, fmt.Errorf("failed to create GitRepository resource: %w", err)
 	}
 
-	objs = append(objs, gr, hr)
-	return objs, nil
+	var helmRes *helmv2.HelmRelease
+	var kustomizeRes *kustomizev1.Kustomization
+
+	switch kind := p.definition.Spec.Artifacts[0].Kind; kind {
+	case "HelmChart":
```

Comment:
do we want to const or type these kinds?

Reply:
That's being done as part of @Skarlso PR 😃